In [7]:
import pandas as pd
import numpy as np

artists = pd.read_csv("billboard_number_one_artists.csv")
songs = pd.read_csv("billboard_number_ones.csv")

In [8]:
# Take the top 10 artists by total weeks spent at #1 (the true chart giants),
# then rank *those 10* by hits-per-week to see who got there efficiently vs. by camping on fewer songs.
top10_weeks = artists.nlargest(10, 'Total_Weeks_At_Number_One').copy()
top10_weeks['Hits_per_Week'] = top10_weeks['Number_Ones'] / top10_weeks['Total_Weeks_At_Number_One']
top10_weeks['Avg_Weeks_per_Hit'] = top10_weeks['Total_Weeks_At_Number_One'] / top10_weeks['Number_Ones']

top10_weeks[['Artist', 'Number_Ones', 'Total_Weeks_At_Number_One', 'Hits_per_Week', 'Avg_Weeks_per_Hit']] \
    .round(2).sort_values('Hits_per_Week', ascending=False).reset_index(drop=True)

,Artist,Number_Ones,Total_Weeks_At_Number_One,Hits_per_Week,Avg_Weeks_per_Hit
0,Madonna,12,32,0.38,2.67
1,Michael Jackson,11,30,0.37,2.73
2,Whitney Houston,11,31,0.35,2.82
3,The Beatles,19,54,0.35,2.84
4,Taylor Swift,13,46,0.28,3.54
5,Mariah Carey,16,82,0.20,5.12
6,Drake,6,33,0.18,5.50
7,Adele,5,34,0.15,6.80
8,Boyz II Men,4,34,0.12,8.50
9,The Black Eyed Peas,3,28,0.11,9.33


### Takeaway

**Madonna** has the best ratio at **0.38** #1 hits per week (about 2.7 weeks per hit), with The Beatles close behind at 0.35 hits/week (about 2.8 weeks per hit). In contrast, Black Eyed Peas averaged just 0.11 hits/week (9.3 weeks at #1 per hit), showing they relied on fewer songs with much longer runs.